techniques/applications/packageManagement/1.3/packageManagement.st

```diff
@@ -0,0 +1,161 @@
+
+      # Version
+      "version_description[${index_pkg}]" string     => "in latest available version",
+                                          ifvarclass => "version_latest_${index_pkg}.(state_present_${index_pkg}|state_upgrade_only_${index_pkg})";
+      "version_description[${index_pkg}]" string     => "in any version",
+                                          ifvarclass => "!version_latest_${index_pkg}.!version_specified_${index_pkg}";
+      "version_description[${index_pkg}]" string     => "in version ${version[${index_pkg}]}",
+                                          ifvarclass => "version_specified_${index_pkg}";
+      # Non compatibility check is done by the "package_state_options" this is just to inform the user
+      "version_description[${index_pkg}]" string     => "in latest available version is not supported and",
+                                          ifvarclass => "version_latest_${index_pkg}.!(state_present_${index_pkg}|state_upgrade_only_${index_pkg})";
+
+
+  classes:
+    any::
+      "state_upgrade_only_${index_pkg}"     expression => strcmp("${state[${index_pkg}]}", "upgrade_only");
+      "state_present_${index_pkg}"          expression => strcmp("${state[${index_pkg}]}", "present");
+
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+    pass1::
+      # This class is different from the one in ncf (where a version number and latest are considered "specified")
+      "version_specified_${index_pkg}"      expression => strcmp("${version[${index_pkg}]}", "specific");
+      "architecture_specified_${index_pkg}" expression => strcmp("${architecture[${index_pkg}]}", "specific");
+      "posthook_specified_${index_pkg}"            not => strcmp("${posthook[${index_pkg}]}", "");
+      "manager_specified_${index_pkg}"             not => strcmp("${manager[${index_pkg}]}", "default");
+      "version_latest_${index_pkg}"         expression => strcmp("${version[${index_pkg}]}", "latest");
+      "allow_unstrusted_${index_pkg}"       expression => strcmp("${manager_allow_untrusted[${index_pkg}]}", "true");
+      "manager_apt_${index_pkg}"            expression => strcmp("${manager[${index_pkg}]}", "apt");
+      "apt_allow_untrusted_${index_pkg}"    expression => "allow_unstrusted_${index_pkg}.((debian.!manager_specified_${index_pkg})|manager_apt_${index_pkg})";
+      "already_installed_${index_pkg}"      expression => "${inner_already_installed_class_prefix_${index_pkg}}_kept";
+
+  methods:
+    pass1::
+      # Force to audit mode to verify if packages are already installed (whatever their version is)
+      "remove_dry_run_mode"            usebundle => push_dry_run_mode("true");
+      "already_installed_${index_pkg}" usebundle => package_state_options("${package[${index_pkg}]}", "any", "${architecture[${index_pkg}]}", "${manager[${index_pkg}]}", "present", "${manager_final_options[${index_pkg}]}"),
```

Comment:
if we have two different technique, one with upgrade only first and the second with install, it will skip the install part and fail